### ISSUE TITLE
no-restricted-imports should have option to allow import * as Smthng from...

### ISSUE THREAD

**Issue by villesau** [eslint/eslint#10778]
**What rule do you want to change?**
no-restricted-imports
**Does this change cause the rule to produce more or fewer warnings?**
Depends on the implementation. If taken as is, fewer. If implemented as new rule, same amount with current setup
**How will the change be implemented? (New option, new default behavior, etc.)?**
It does not matter. Important is that such rule should exist.
**current rule example**

```json
    "no-restricted-imports": ["error", {
      "name": "react",
      "importNames": ["default"],
      "message": "use import * as React insetad"
    }],
```

**What does the rule currently do for this code?**

```js
import * as React from 'react'; //disallowed
```
```js
import React from 'react'; //disallowed
```
**What will the rule do after it's changed?**

```js
import React from 'react'; //disallowed
```

```js
import * as React from 'react'; //allowed
```

***Why this change is needed?***

When using static type checker like Flow, above imports behave differently: `import * as React` returns types, when `import React` returns only the default export. They also have slight semantic difference: `import * as` returns an object containing all the export, when `import React` returns the default export., what ever it might be. This means that also only default import should be bannable. Now both are with the setup above.

Here is Flow documentation about this ( https://flow.org/en/docs/react/components/ ):

> Note: We import React as a namespace here with import * as React from 'react' instead of as a default with import React from 'react'. When importing React as an ES module you may use either style, but importing as a namespace gives you access to React’s utility types.
